pkg/pb/synthetic_monitoring/checks_extra.go

```diff
@@ -147,14 +150,22 @@ func (c *Check) Validate() error {
 	}
 
 	// frequency must be in [1, 120] seconds
-	if c.Frequency < 1*1000 || c.Frequency > 120*1000 {
+	if c.Type() != CheckTypeTraceroute && (c.Frequency < 1*1000 || c.Frequency > 120*1000) {
 		return ErrInvalidCheckFrequency
 	}
 
 	// timeout must be in [1, 10] seconds, and it must be less than
 	// frequency (otherwise we can end up running overlapping
 	// checks)
-	if c.Timeout < 1*1000 || c.Timeout > 10*1000 || c.Timeout > c.Frequency {
+	if c.Type() == CheckTypeTraceroute && c.Timeout > 10*3000 {
+		return ErrInvalidCheckTimeout
+	}
+
+	if c.Type() != CheckTypeTraceroute && c.Timeout > 10*1000 {
+		return ErrInvalidCheckTimeout
+	}
+
+	if c.Timeout < 1*1000 || c.Timeout > c.Frequency {
 		return ErrInvalidCheckTimeout
 	}
```

Comment:
A couple of questions on this part:
- Could we add a few more comments to explain why you've chosen to set up the validations this way?
- I'm wondering whether we could extract the complexity out here somehow, and simplify (compose?) the code a bit more here? I'm thinking we could have a `validate_traceroute` function, separate from a `validate_bbe_check` function (or something similar), in order to have fewer places where we are doing `c.Type() == CheckTypeTraceroute` or `c.Type() =! CheckTypeTraceroute`

Reply:
🤔 This is a good idea

Reply:
I pulled out timeout and frequency validations into their own functions as a first step (also because this function was running into a "cyclomatic complexity" linting rule)